### PR TITLE
Bump ts-node to 10.9, allow node 18, fix json import assert

### PR DESCRIPTION
**Pull request by birkskyum**
Compatibility with node 16 and 18 interchangeably.

Changes:
- Ts-node is bumped to 10.9 - that cause that we needed to add some `import assert type json` statements.
- Jest couldn't manage that, so now instead of having the jest tests do type checking, they only execute the code, and a separate `typecheck` command is now package.json instead using tsc, and also a github test runner for the CI/CD. This is actually better as theres no point in running a full typecheck for all 122 unit tests...
- Rollup couldn't handle the json import statements either, so a new [plugin was added for that](https://www.npmjs.com/package/rollup-plugin-import-assert).

### CODE REVIEW

**Comment by birkskyum** [comment]
There is a bit of work left related to the unassert plugin. The inline assert tests take up 5 kb, so if we leave them in production they add that. We can also remove them altogether, or we figure out how fix the unassert plugin.

**Comment by HarelM** [comment]
I don't see a lot of value in asserting inside the code. The tests should cover the cases where assert is needed in my opinion...

**Comment by birkskyum** [comment]
I actually managed to fix the plugin. I made an upsteam PR a week ago https://github.com/unassert-js/rollup-plugin-unassert/pull/8, but for now we can just use the fork i made.

**Comment by birkskyum** [comment]
So now we have everything working. I also think we should remove these asserts at some point, as they are starting to hold us back in different ways. There is a branch here called [remove-browserify](https://github.com/maplibre/maplibre-gl-js/compare/main...remove-browserify) where I commented all the node assert

File: .github/workflows/typecheck.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    timeout-minutes: 20
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js 16 x64
+        uses: actions/setup-node@v2
+        with:
+          node-version: 16
+          architecture: x64
+      - run: npm ci
+      - run: npm run typecheck
```
Comment (HarelM) [inline .github/workflows/typecheck.yml:21]: Does this run tsc on the tests files as well?

**Review by HarelM — APPROVED** [review]
Seems like a lot of dependencies were added... While I believe this is needed we need to think how to reduce those in the future...

**Comment by birkskyum** [comment]
> Seems like a lot of dependencies were added... While I believe this is needed we need to think how to reduce those in the future...

Yes, 2 dependencies for rollup to recognise the import assert. I expect it'll somehow be baked into rollup/typescript quite soon so we can remove them again.

File: .github/workflows/typecheck.yml (hunk 1 above)
Comment (birkskyum) [reply .github/workflows/typecheck.yml:21]: Good catch - it didn't. I have just now added a tsconfig.typecheck.json file so we can tailor the typecheck specifically to the areas we want.

**Pull request merged**